sklearn/utils/graph.py

```diff
@@ -21,34 +21,38 @@
 # Path and connected component analysis.
 # Code adapted from networkx
 def single_source_shortest_path_length(graph, source, *, cutoff=None):
-    """Return the shortest path length from source to all reachable nodes.
-
-    Returns a dictionary of shortest path lengths keyed by target.
+    """Return the length of the shortest path from source to all reachable nodes.
 
     Parameters
     ----------
-    graph : {sparse matrix, ndarray} of shape (n, n)
+    graph : {sparse matrix, ndarray} of shape (n_nodes, n_nodes)
         Adjacency matrix of the graph. Sparse matrix of format LIL is
         preferred.
 
     source : int
-       Starting node for path.
+       Start node for path.
 
     cutoff : int, default=None
         Depth to stop the search - only paths of length <= cutoff are returned.
 
+    Returns
+    -------
+    paths : dict
+        Reachable end nodes mapped to lenght of path from source,
+        i.e. `{end: path_length}`.
+
     Examples
     --------
     >>> from sklearn.utils.graph import single_source_shortest_path_length
     >>> import numpy as np
     >>> graph = np.array([[ 0, 1, 0, 0],
     ...                   [ 1, 0, 1, 0],
-    ...                   [ 0, 1, 0, 1],
-    ...                   [ 0, 0, 1, 0]])
-    >>> list(sorted(single_source_shortest_path_length(graph, 0).items()))
-    [(0, 0), (1, 1), (2, 2), (3, 3)]
+    ...                   [ 0, 1, 0, 0],
+    ...                   [ 0, 0, 0, 0]])
+    >>> single_source_shortest_path_length(graph, 0)
+    {0: 0, 1: 1, 2: 2}
     >>> graph = np.ones((6, 6))
-    >>> list(sorted(single_source_shortest_path_length(graph, 2).items()))
+    >>> sorted(single_source_shortest_path_length(graph, 2).items())
```

Comment:
Why did you change the example ?

Reply:
I felt that the first example should show a dictionary, not a dict on `.items()` is called and then sorted. Also including an unreachable node makes it clear that unreachable nodes do not show up in the result.

Second example I deleted the `list()` constructor since `sorted` already returns a list.

Reply:
makes sense. I'm fine with this change. Thanks